app/controllers/api/v1/documents.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'doorkeeper/grape/helpers'
+
+module API
+  module V1
+    class Documents < Grape::API
+      helpers Doorkeeper::Grape::Helpers
+
+      before do
+        doorkeeper_authorize!
+
+        def current_account
+          Account.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
+        end
+      end
+
+      desc 'Account related routes'
+      resource :document do
+        desc 'Return current user documents list'
+        get '/' do
+          current_account.profile.documents.as_json(only: %i[upload doc_type doc_number doc_expire])
+        end
+
+        desc 'Return information about current resource owner'
+        post '/upload' do
+          profile = current_account.profile
+          return error!('Can\'t upload document, create profile first', 400) unless profile
+          document = Document.new
```

Comment:
Maybe Document.new(params) but make sure we filter params with a whitelist list they do in rails